### PR TITLE
Add CUDA 11.8

### PR DESCRIPTION
**Pull request by bdice**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Adds CUDA 11.8. Depends on https://github.com/conda-forge/docker-images/pull/217, https://github.com/conda-forge/conda-forge-ci-setup-feedstock/pull/207. cc: @m3vaz

Following the example of #77, #83. Instructions here: https://conda-forge.org/docs/maintainer/knowledge_base.html#adding-support-for-a-new-cuda-version

### CODE REVIEW

**Comment by bdice** [comment]
@conda-forge-admin, please rerender

**Review by jaimergp — CHANGES_REQUESTED** [review]
Please do not merge yet. Thanks!

**Comment by jakirkham** [comment]
@conda-forge-admin, please re-render

**Review by jakirkham — APPROVED** [review]

**Comment by bdice** [comment]
@conda-forge-admin, please re-render

**Comment by bdice** [comment]
@conda-forge-admin, please re-render

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -24,7 +24,7 @@ cuda_compiler_version:
   - None
 
   - 9.2    # [linux64]
-  - 10.0   # [linux64 or win]
```
Comment (jakirkham) [inline recipe/conda_build_config.yaml:27]: Windows CUDA 10.0 is dropped here as it is incompatible with `vs2019`. Given there are already packages for it and conda-forge has largely dropped CUDA 10.0, we are skipping Windows CUDA 10.0 here

**Comment by jakirkham** [comment]
The Windows CUDA 11.8 build had failed due to a checksum issue. Summarized the problem here ( https://github.com/conda-forge/conda-forge-ci-setup-feedstock/pull/216#discussion_r1037513019 ) and submitted a fix ( https://github.com/conda-forge/conda-forge-ci-setup-feedstock/pull/218 ) that has since been merged. Restarting CI to clear that issue out

**Comment by jakirkham** [comment]
Thanks Bradley! 🙏

**Pull request merged**